tests/tests.h

```diff
@@ -182,6 +182,29 @@ void sort_file_contents (const std::string &filename)
 }
 
 
+/*
+ * simple ADLER32 checksum for a range of chars
+ */
+template <class IT>
+unsigned int checksum(const IT &begin, const IT &end)
+{
+  AssertThrow(sizeof(unsigned int)==4, ExcInternalError());
+
+  unsigned int a = 1;
+  unsigned int b = 0;
+
+  IT it = begin;
+
+  while (it != end)
+    {
+      a = (a + (unsigned char)*it) % 65521;
```

Comment:
Yes, that is ok as well. We can always extend that later.

(Looping over the bytes of the returned object is only correct for POD types anyway, and even then most easily done if you have `auto` types available. Both benefit from C++11, so let's leave the extension to other types for a later time when we actually need it.)